File: comments/src/main/java/net/particify/arsnova/comments/exception/BadRequestException.java
package net.particify.arsnova.comments.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.server.ResponseStatusException;

public class BadRequestException extends ResponseStatusException {

  public BadRequestException() {
    super(HttpStatus.BAD_REQUEST);
  }

  public BadRequestException(final String message) {
    super(HttpStatus.BAD_REQUEST, message);
  }

  public BadRequestException(final String message, final Throwable e) {
    super(HttpStatus.BAD_REQUEST, message, e);
  }
}

File: authz/src/main/kotlin/de/thm/arsnova/service/authservice/exception/BadRequestException.kt
package de.thm.arsnova.service.authservice.exception

import org.springframework.http.HttpStatus
import org.springframework.web.server.ResponseStatusException

class BadRequestException : ResponseStatusException {
  constructor() : super(HttpStatus.BAD_REQUEST)
  constructor(reason: String?) : super(HttpStatus.BAD_REQUEST, reason)
  constructor(reason: String?, cause: Throwable?) : super(HttpStatus.BAD_REQUEST, reason, cause)
}
